meson.build

```diff
@@ -0,0 +1,173 @@
+project(
+    'cloud-init',
+    version: run_command('./tools/read-version', check: true).stdout().strip(),
+    meson_version: '>=0.63.0',  # rockylinux/9
+    default_options: [
+      # The default can yield broken results.
+      'python.install_env=auto'
+    ]
+)
+completions = dependency('bash-completion')
+systemd = dependency('systemd')
+system = host_machine.system()
+
+libdir = get_option('libdir')
+pkgdatadir = get_option('datadir')
+sysconfdir = get_option('sysconfdir')
+INIT_SYSTEM = get_option('init_system')
+VALID_INIT_SYSTEMS = [
+  'systemd', 'sysvinit', 'sysvinit_deb', 'sysvinit_dragonfly',
+  'sysvinit_freebsd', 'sysvinit_netbsd', 'sysvinit_openbsd'
+]
+if not VALID_INIT_SYSTEMS.contains(INIT_SYSTEM)
+  error(
+    'Invalid init_system supplied: "@0@". Must be one of: @1@'.format(
+      INIT_SYSTEM, VALID_INIT_SYSTEMS)
+  )
+endif
+
+
+lib_exec_dir = join_paths('/usr', get_option('libexecdir'), 'cloud-init')
+
+pymod = import('python')
+python = pymod.find_installation('python3')
+
+find = find_program('find')
+
+install_subdir(
+  'cloudinit', install_dir : python.get_install_dir(), install_tag: 'python')
+
+# Binaries and script entrypoints
+bash_completions_dir = completions.get_variable(
+  pkgconfig: 'completionsdir', default_value: '/etc/bash_completion.d')
+install_data(
+  'bash_completion/cloud-init',
+  install_dir: bash_completions_dir, install_tag: 'scripts')
+
+install_data(
+  ['tools/ds-identify', 'tools/hook-hotplug','tools/uncloud-init', 'tools/write-ssh-key-fingerprints' ],
+  install_dir: lib_exec_dir,  install_tag: 'scripts')
+
+
+install_data(
+  ['tools/cloud-init-per', 'scripts/cloud-id', 'scripts/cloud-init'],
+  install_mode: 'rwxr-xr-x',
+  install_dir: get_option('bindir'),
+)
```

Comment:
Stylistically, I prefer closing multi-line functions with parenthesis on a new line, like this function call does. Can we please do this consistently throughout this file?

Reply:
As an aside, I use the mesonlsp LSP to format meson files. If we agree to using it to format the code, we shouldn't need to nitpick formatting like this. I did just notice that the upstream project was recently archived, but it still seems to do a nice job.

Reply:
I think we've done the lion's share of formatting discussions on this PR and have established conventions that we can follow from this point forward. If we find ourselves revisiting these formatting issues much in the future, then we can consider adding automated formatting checks or a local policy to address this in pre-commit hooks etc. As it is, I'm hoping since we only have two meson* files in the project, we can adhere to prior art without a lot of back and forth or tooling changes.